### PR TITLE
Add some test specs & targets for e2e tests

### PR DESCRIPTION
**Pull request by smukherj1**
First step in setting up e2e tests for automated toolchain configs
release

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -131,6 +131,49 @@ rbe_autoconfig(
...
+load(
+    "//configs/tests/dependency-tracking:ubuntu1604_no_updates.bzl",
+    _ubuntu1604_bazel_no_updates = "bazel",
+    _ubuntu1604_configs_version_no_updates = "configs_version",
+    _ubuntu1604_digest_no_updates = "digest",
+    _ubuntu1604_registry_no_updates = "registry",
+    _ubuntu1604_repository_no_updates = "repository",
+)
+
+# Automatic config generation target for RBE Ubuntu 16.04
```
Comment (nlopezgi) [inline WORKSPACE]: change the comment as it reads the same as the one in line 124 and this has a different purpose

File: WORKSPACE [hunk 2]
```diff
@@ -131,6 +131,49 @@ rbe_autoconfig(
...
+rbe_autoconfig(
+    name = "rbe_autoconfig_autogen_ubuntu1604_test_no_updates",
```
Comment (nlopezgi) [inline WORKSPACE]: call this "no_update_test" (or something that is more reflective of the purpose, not the implementation detail)
also, use that (or some other) shorter prefix/suffic for "_ubuntu1604_xxx_no_updates"

File: WORKSPACE [hunk 3]
```diff
@@ -131,6 +131,49 @@ rbe_autoconfig(
...
+    _ubuntu1604_bazel_updates_exist = "bazel",
+    _ubuntu1604_configs_version_updates_exist = "configs_version",
+    _ubuntu1604_digest_updates_exist = "digest",
+    _ubuntu1604_registry_updates_exist = "registry",
+    _ubuntu1604_repository_updates_exist = "repository",
+)
+
+# Automatic config generation target for RBE Ubuntu 16.04
+rbe_autoconfig(
+    name = "rbe_autoconfig_autogen_ubuntu1604_test_updates_exist",
```
Comment (nlopezgi) [inline WORKSPACE]: rename to "trigger_test"

File: WORKSPACE [hunk 4]
```diff
@@ -131,6 +131,49 @@ rbe_autoconfig(
...
+load(
+    "//configs/tests/dependency-tracking:ubuntu1604_updates_exist.bzl",
+    _ubuntu1604_bazel_updates_exist = "bazel",
+    _ubuntu1604_configs_version_updates_exist = "configs_version",
+    _ubuntu1604_digest_updates_exist = "digest",
+    _ubuntu1604_registry_updates_exist = "registry",
+    _ubuntu1604_repository_updates_exist = "repository",
+)
+
+# Automatic config generation target for RBE Ubuntu 16.04
```
Comment (nlopezgi) [inline WORKSPACE]: Update docs

File: configs/tests/README.md [hunk 5]
```diff
@@ -0,0 +1,2 @@
+This directory has YAML config files used to run integration tests
```
Comment (nlopezgi) [inline configs/tests/README.md]: move to under tests/config

File: WORKSPACE (hunk 1 above)
Comment (smukherj1) [reply WORKSPACE]: Done

File: WORKSPACE (hunk 2 above)
Comment (smukherj1) [reply WORKSPACE]: Done

File: WORKSPACE (hunk 4 above)
Comment (smukherj1) [reply WORKSPACE]: Done

File: configs/tests/README.md (hunk 5 above)
Comment (smukherj1) [reply configs/tests/README.md]: Done

File: WORKSPACE (hunk 3 above)
Comment (smukherj1) [reply WORKSPACE]: Done

File: tests/config/dependency-tracking/ubuntu1604_updates_exist.bzl [hunk 6]
```diff
@@ -0,0 +1,6 @@
+"""Information tracking the latest published configs."""
+bazel = "0.24.0"
+registry = "gcr.io"
+repository = "asci-toolchain/toolchain-configs-e2e-test/rbe-ubuntu16-04"
+digest = "sha256:thisisafakeshathatisboundtogetupdatedhuehuehue"
```
Comment (nlopezgi) [inline tests/config/dependency-tracking/ubuntu1604_updates_exist.bzl]: r/thisisafakeshathatisboundtogetupdatedhuehuehue/<placeholder to update>

File: WORKSPACE [hunk 7]
```diff
@@ -131,6 +131,50 @@ rbe_autoconfig(
...
+    digest = _ubuntu1604_digest_no_updates,
+    output_base = "configs/ubuntu16_04_clang/{}".format(_ubuntu1604_configs_version_no_updates),
+    registry = _ubuntu1604_registry_no_updates,
+    repository = _ubuntu1604_repository_no_updates,
+    use_checked_in_confs = False,
+)
+
+load(
+    "//tests/config/dependency-tracking:ubuntu1604_updates_exist.bzl",
+    _ubuntu1604_bazel_updates_exist = "bazel",
```
Comment (nlopezgi) [inline WORKSPACE]: name this 
_bazel_trigger_config_gen
_digest_trigger_config_gen

Also name the file 
//tests/config/dependency-tracking:trigger_config_gen.bzl

Similar for the one above, file should be
//tests/config/dependency-tracking:no_updates.bzl

File: tests/config/dependency-tracking/ubuntu1604_updates_exist.bzl (hunk 6 above)
Comment (smukherj1) [reply tests/config/dependency-tracking/ubuntu1604_updates_exist.bzl]: Done

File: WORKSPACE (hunk 7 above)
Comment (smukherj1) [reply WORKSPACE]: Done

File: tests/config/dependency-specs/ubuntu16_04_no_updates.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# Location of the config-dependency tracker file on the repository
```
Comment (nlopezgi) [inline tests/config/dependency-specs/ubuntu16_04_no_updates.yaml]: ubuntu16_04_no_update.yaml

File: tests/config/dependency-specs/ubuntu16_04_updates_exist.yaml [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# Location of the config-dependency tracker file on the repository
```
Comment (nlopezgi) [inline tests/config/dependency-specs/ubuntu16_04_updates_exist.yaml]: ubuntu16_04_trigger_config_gen.yaml

File: tests/config/dependency-specs/ubuntu16_04_no_updates.yaml (hunk 8 above)
Comment (smukherj1) [reply tests/config/dependency-specs/ubuntu16_04_no_updates.yaml]: Done

File: tests/config/dependency-specs/ubuntu16_04_updates_exist.yaml (hunk 9 above)
Comment (smukherj1) [reply tests/config/dependency-specs/ubuntu16_04_updates_exist.yaml]: Done

File: tests/config/dependency-tracking/BUILD [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+licenses(["notice"])  # Apache 2.0
+
+package(default_visibility = ["//visibility:public"])
+
+bzl_library(
+    name = "dependency-tracking",
+    srcs = [
+        "ubuntu1604_no_updates.bzl",
+        "ubuntu160_updates_exist.bzl",
```
Comment (nlopezgi) [inline tests/config/dependency-tracking/BUILD]: update

File: tests/config/dependency-specs/ubuntu16_04_trigger_config_gen.yaml [hunk 11]
```diff
@@ -0,0 +1,14 @@
+# Location of the config-dependency tracker file on the repository
+# specified in the Github spec YAML file to the toolchain configs update
+# service.
+revisionsFilePath: "configs/tests/dependency-tracking/ubuntu1604_updates_exist.bzl"
```
Comment (nlopezgi) [inline tests/config/dependency-specs/ubuntu16_04_trigger_config_gen.yaml]: update

File: tests/config/dependency-tracking/BUILD (hunk 10 above)
Comment (smukherj1) [reply tests/config/dependency-tracking/BUILD]: Done

File: tests/config/dependency-specs/ubuntu16_04_trigger_config_gen.yaml (hunk 11 above)
Comment (smukherj1) [reply tests/config/dependency-specs/ubuntu16_04_trigger_config_gen.yaml]: Done

**Pull request merged**